src/features/tempRole/temporaryRoleCommand.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  ApplicationCommandOptionType,
+  ApplicationCommandType,
+  GuildMember,
+  PermissionsBitField,
+  Role,
+  time,
+} from 'discord.js'
+
+import parse from 'parse-duration'
+
+import { prisma } from '@/prisma'
+import { defineCommand } from '@/types/defineCommand'
+
+export const temporaryRoleCommand = defineCommand({
+  data: {
+    name: 'temp-role',
+    description: 'Set a temporary role for a user',
+    defaultMemberPermissions: PermissionsBitField.Flags.ManageRoles,
+    type: ApplicationCommandType.ChatInput,
+    options: [
+      {
+        name: 'user',
+        description: 'User to set the temporary role for',
+        type: ApplicationCommandOptionType.User,
+        required: true,
+      },
+      {
+        name: 'role',
+        description: 'Role to set',
+        type: ApplicationCommandOptionType.Role,
+        required: true,
+      },
+      {
+        name: 'duration',
+        description: 'Duration of the temporary role e.g. 1d',
+        type: ApplicationCommandOptionType.String,
+        required: true,
+      },
+    ],
+  },
+  ephemeral: true,
+  execute: async (botContext, interaction) => {
+    if (!interaction.guild || !interaction.isChatInputCommand()) return
+
+    // Get user from the interaction
+    const user = interaction.options.getMember('user') as GuildMember
+
+    // Get role from the interaction
+    const role = interaction.options.getRole('role') as Role
+
+    // Get duration from the interaction
+    const duration = interaction.options.getString('duration') as string
+    const durationMs = parse(duration)
+
+    if (durationMs === null) {
+      await interaction.editReply({
+        content: `Invalid duration: ${duration}`,
+      })
+      return
+    }
+
+    // Check if the user already has the role
+    if (user.roles.cache.has(role.id)) {
+      await interaction.editReply({
+        content: `User: ${user} already has role: ${role}`,
+      })
+      return
+    }
+
+    const expriesAt = new Date(Date.now() + durationMs)
+
+    // Add the role to the user
+    user.roles.add(role)
```

Comment:
Is it necessary?
in djs guide: https://discordjs.guide/popular-topics/faq.html#how-do-i-add-a-role-to-a-guild-member

Reply:
You kinda have to anyway because you should include try catch for potential error.
https://github.com/kaogeek/kaogeek-discord-bot/pull/155#discussion_r1207680762

I guess the guide have no await because they have to make it somewhat general (there no telling that it async function or not)

From my search on stackoverflow, youtube. Discord.js guide rarely use await, what happens, I don't know.

Reply:
the rule of thumb is: hover over the method e.g. `add`. if the return type says that it is a promise, generally you want to add await.

> From my search on stackoverflow, youtube. Discord.js guide rarely use await, what happens, I don't know.

it seems that Discord.js guide was written mostly for people who are new to JavaScript, who don't have a strong grasp of Promise or async code.